Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -0,0 +1,1076 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Display;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Helpers;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Parse;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// An efficient and extensible control that can parse and render markdown.
+    /// </summary>
+    public sealed partial class MarkdownTextBlock : Control, ILinkRegister
+    {
+        /// <summary>
+        /// Holds a list of hyperlinks we are listening to.
+        /// </summary>
+        private List<Hyperlink> _listeningHyperlinks = new List<Hyperlink>();
+
+        /// <summary>
+        /// The root element for our rendering.
+        /// </summary>
+        private Border _rootElement = null;
+
+        /// <summary>
+        /// Fired when the text is done parsing and formatting. Fires each time the markdown is rendered.
+        /// </summary>
+        public event EventHandler<OnMarkdownRenderedArgs> OnMarkdownRendered;
```

Comment:
It can be used to listen for when a large, complex, block of markdown is done. So you can have loading text or whatever.